game/src/gui/foundation/key_event.h

```diff
@@ -11,6 +11,10 @@ class KeyEvent {
  public:
   enum class State : uint16_t { kIdle, kDown, kUp };
   enum class Mod : uint16_t { kNone = 0, kCtrl = 1, kShift = 2, kAlt = 4 };
+  enum {
```

Comment:
Please make it to be `enum class Key`. The name also can be something better than just `Key`.

Reply:
Had compile error when used with `enum class Code { ESC=27; };`
```
    switch (e.GetCode()) {
      case foundation::KeyEvent::ESC:
```
Do you know how to solve this?